File: profiles/nodejs-ts/template/tests/hello.test.ts
import { describe, expect, it } from 'vitest';
import { EmptyNameError, greet } from '../src/hello.js';

describe('greet', () => {
  it('returns a formatted greeting for a plain name', () => {
    expect(greet('world')).toBe('Hello, world!');
  });

  it('trims leading and trailing whitespace', () => {
    expect(greet('  Ada  ')).toBe('Hello, Ada!');
  });

  it('throws EmptyNameError on empty string', () => {
    expect(() => greet('')).toThrow(EmptyNameError);
  });

  it('throws EmptyNameError on whitespace-only string', () => {
    expect(() => greet('   \t\n')).toThrow(EmptyNameError);
  });

  it('EmptyNameError has the expected name', () => {
    try {
      greet('');
    } catch (err) {
      expect(err).toBeInstanceOf(EmptyNameError);
      expect((err as Error).name).toBe('EmptyNameError');
    }
  });
});

File: profiles/java/template/src/test/java/_pkg/HelloTest.java
package {{ java_package }};

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

class HelloTest {

  @Test
  @DisplayName("greet returns formatted greeting for a plain name")
  void greetReturnsGreeting() {
    assertEquals("Hello, world!", Hello.greet("world"));
  }

  @Test
  @DisplayName("greet strips leading and trailing whitespace")
  void greetStripsWhitespace() {
    assertEquals("Hello, Ada!", Hello.greet("  Ada  "));
  }

  @Test
  @DisplayName("greet rejects null name")
  void greetRejectsNull() {
    assertThrows(NullPointerException.class, () -> Hello.greet(null));
  }

  @Test
  @DisplayName("greet rejects empty name")
  void greetRejectsEmpty() {
    assertThrows(IllegalArgumentException.class, () -> Hello.greet(""));
  }

  @Test
  @DisplayName("greet rejects whitespace-only name")
  void greetRejectsWhitespace() {
    assertThrows(IllegalArgumentException.class, () -> Hello.greet("   \t\n"));
  }
}
